src/include/catalog/schema.h

```diff
@@ -0,0 +1,107 @@
+#pragma once
+#include <string>
+#include <utility>
+#include <vector>
+#include "common/constants.h"
+#include "common/macros.h"
+#include "common/typedefs.h"
+#include "storage/storage_defs.h"
+#include "type/type_id.h"
+#include "type/type_util.h"
+
+namespace terrier::catalog {
+
+/**
+ * Internal object for representing SQL table schema. Currently minimal until we add more features to the system.
+ * TODO(Matt): we should make sure to revisit the fields and their uses as we bring in a catalog to replace some of the
+ * reliance on these classes
+ */
+class Schema {
+ public:
+  /**
+   * Internal object for representing SQL table column. Currently minimal until we add more features to the system.
+   * TODO(Matt): we should make sure to revisit the fields and their uses as we bring in a catalog to replace some of
+   * the reliance on these classes
+   */
+  class Column {
+   public:
+    /**
+     * Instantiates a Column object, primary to be used for building a Schema object
+     * @param name column name
+     * @param type SQL type for this column
+     * @param nullable true if the column is nullable, false otherwise
+     * @param oid internal unique identifier for this column
+     */
+    Column(std::string name, const type::TypeId type, const bool nullable, const col_oid_t oid)
+        : name_(std::move(name)),
+          type_(type),
+          attr_size_(type::TypeUtil::GetTypeSize(type_)),
+          nullable_(nullable),
+          inlined_(true),
+          oid_(oid) {
+      if (attr_size_ == 0) {
+        // this is a varlen attribute
+        attr_size_ = 8;
+        inlined_ = false;
+      }
+      TERRIER_ASSERT(attr_size_ == 1 || attr_size_ == 2 || attr_size_ == 4 || attr_size_ == 8,
+                     "Attribute size must be 1, 2, 4, or 8 bytes.");
+      TERRIER_ASSERT(type_ != type::TypeId::INVALID, "Attribute type cannot be INVALID.");
+    }
+    /**
+     * @return column name
+     */
+    const std::string &GetName() const { return name_; }
+    /**
+     * @return true if the column is nullable, false otherwise
+     */
+    bool GetNullable() const { return nullable_; }
+    /**
+     * @return size of the attribute in bytes
+     */
+    uint8_t GetAttrSize() const { return attr_size_; }
+    /**
+     * @return true if the attribute is inlined, false if it's a pointer to a varlen entry
+     */
+    bool GetInlined() const { return inlined_; }
+    /**
+     * @return SQL type for this column
+     */
+    type::TypeId GetType() const { return type_; }
+
+   private:
+    const std::string name_;
+    const type::TypeId type_;
+    uint8_t attr_size_;
+    const bool nullable_;
+    bool inlined_;
+    const col_oid_t oid_;
+    // TODO(Matt): default value would go here
+    // Value default_;
```

Comment:
Should this Column also provide mapping to col_id_t in the storage layer?

Reply:
Possibly! We don't have a catalog yet. I almost think the mapping from col_oit_t to col_id should exist at the SqlTable though, where we're handle stuff like multiple layouts...